services/meta/client.go

```diff
@@ -957,35 +957,6 @@ func (c *Client) updateAuthCache() {
 	c.authCache = newCache
 }
 
-// snapshot will save the current meta data to disk
-func snapshot(path string, data *Data) error {
-	file := filepath.Join(path, metaFile)
-	tmpFile := file + "tmp"
-
-	f, err := os.Create(tmpFile)
-	if err != nil {
-		return err
-	}
-	defer f.Close()
-
-	var d []byte
-	if b, err := data.MarshalBinary(); err != nil {
-		return err
-	} else {
-		d = b
-	}
-
-	if _, err := f.Write(d); err != nil {
-		return err
-	}
-
-	if err = f.Close(); nil != err {
-		return err
-	}
-
-	return os.Rename(tmpFile, file)
```

Comment:
Instead of copying the entire function, can you just make a private function `renameFile` so only that part of the code has to be in a separate file? I think this change will not work on Mac OS X because the function will only exist on Linux or Windows. I think renaming that file to something like `file_unix.go` works correctly.